### PR TITLE
eshell: set eshell-path-env in activate and deactivate

### PR DESCRIPTION
**Pull request by seanfarley**
It would seem this is the bare minimum to get eshell to recognize the new path
value. I've tested this in many scenarios but would love for others to test and
give feedback!

### CODE REVIEW

File: pyvenv.el [hunk 1]
```diff
@@ -189,7 +190,9 @@ This is usually the base name of `pyvenv-virtual-env'.")
                                 ;; No "=" means to unset
                                 "PYTHONHOME")
                                process-environment)
-          ))
+          )
+    ;; set eshell path to same as exec-path
+    (setq eshell-path-env (mapconcat 'identity exec-path ":")))
```
Comment (jorgenschaefer) [inline pyvenv.el]: You can put this into the other `setq` form as another variable

Comment (jorgenschaefer) [reply pyvenv.el]: Oh, and apparently we do not set this back to the original value on deactivation. Is that intended? (I do not use eshell and do not know the expected behavior)

Comment (seanfarley) [reply pyvenv.el]: Ah, right; I didn't know off the top of my head if a value set in `setq` could be used in the same `setq` in a later line. The previous `exec-path` is restored on line 220 so later in `pyvenv-deactivate` I set `eshell-path-env` to that restored value.

Comment (seanfarley) [reply pyvenv.el]: Should I put the `setq` in the previous statement?

Comment (jorgenschaefer) [reply pyvenv.el]: Two `setq` in sequence do not make sense, so merging the two into a single statement makes sense.

I don't know anything about eshell, so no idea if `eshell-path-env` ever should be different from `exec-path`, but if not, this works for me.

Comment (seanfarley) [reply pyvenv.el]: Oh, d'oh, yeah, that's a good point about `eshell-path-env`. For me, it's always the same but I guess that's not always guaranteed. I'll rework this.

**Comment by seanfarley** [comment]
Ok, I've updated the PR with your feedback :-)

**Comment by jorgenschaefer** [comment]
Thank you very much for your contribution! :-)

**Comment by seanfarley** [comment]
Yay!

**Pull request merged**